src/main/java/com/team766/logging/Logger.java

```diff
@@ -96,15 +100,19 @@ public Collection<LogEntry> recentEntries() {
 	public void logData(final Severity severity, final String format, final Object... args) {
 		var entry = LogEntry.newBuilder().setTime(getTime()).setSeverity(severity)
 				.setCategory(m_category);
-		entry.setMessageStr(String.format(format, args));
+		String message = String.format(format, args);
+		entry.setMessageStr(message);
 		m_recentEntries.add(entry.build());
-		entry.setMessageStr(format);
+		
 		for (Object arg : args) {
 			SerializationUtils.valueToProto(arg, entry.addArgBuilder());
 		}
 		if (m_logWriter != null) {
 			m_logWriter.logStoredFormat(entry);
 		}
+		if (ALSO_LOG_TO_DATALOG) {
+			DataLogManager.log(message);
```

Comment:
does this cause any issues in the simulator, since it's enabled by default?

Reply:
I singled out the added `DataLogManager` calls here because I think the rest of the changes only apply when targeting physical robots.

I think `RobotProvider.instance instanceof WPIRobotProvider` is maybe the "best" way to detect running on a physical robot (unless I've forgotten something, which is entirely possible), but it feels like a code smell (I'm ok with it being tech debt though).

Maybe check first whether this actually fails in the simulator or not - maybe it will Just Work™

Since (AFAIK) we don't have any external customers, I don't have a strong preference for whether this gets added here first or incubated in the 2024 repo.